### PR TITLE
Migrate from test builders to structs

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

This is helpful as we migrate to the v1beta1 types since we won't have
to maintain separate test builders for v1beta1 types.

Ref #1067, #735

/kind cleanup
/release-note-none

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Comment by dibyom** [comment]
/assign @wlynch 

@jmcshane

**Comment by jmcshane** [comment]
Wondering, does this PR allow us to remove any of the existing builders? I feel like moving towards reducing the surface area of the builder functions is the way to go.

**Comment by dibyom** [comment]
> Wondering, does this PR allow us to remove any of the existing builders? 

Perhaps. We'll have to double check with the CLI folks. They might be using the test builders

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go [hunk 1]
```diff
@@ -38,113 +37,276 @@ func Test_EventListenerValidate(t *testing.T) {
...
+		el: &v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "namespace",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				Triggers: []v1alpha1.EventListenerTrigger{{
+					Template: &v1alpha1.EventListenerTemplate{
+						Ref:        ptr.String("dne"),
+						APIVersion: "v1alpha1", // TODO: test with v1alpha1 version as well.
```
Comment (wlynch) [inline pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:49]: Comment doesn't quite make sense (it looks like we're already testing v1alpha1?) - remove or rephrase?

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go [hunk 2]
```diff
@@ -38,113 +37,276 @@ func Test_EventListenerValidate(t *testing.T) {
...
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "namespace",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				Triggers: []v1alpha1.EventListenerTrigger{{
+					Bindings: []*v1alpha1.EventListenerBinding{{
+						Ref:        "tb",
+						Kind:       v1alpha1.NamespacedTriggerBindingKind,
+						APIVersion: "v1alpha1", // TODO: APIVersions seem wrong?
```
Comment (wlynch) [inline pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:95]: TODO is vague - I'm not sure what would need to be done to resolve this. Can you link to an issue / rephrase this to give more detail?

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go [hunk 3]
```diff
@@ -503,72 +610,196 @@ func TestEventListenerValidate_error(t *testing.T) {
...
-						DeprecatedGitLab:    &v1alpha1.GitLabInterceptor{},
-						DeprecatedBitbucket: &v1alpha1.BitbucketInterceptor{},
-					}},
+					Template: &v1alpha1.EventListenerTemplate{Ref: ptr.String(""), APIVersion: "v1alpha1"},
 				}},
 			},
 		},
 	}, {
-		name: "CEL interceptor with no filter or overlays",
+		//	name: "Interceptor Name only",
```
Comment (wlynch) [inline pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:618]: Should we remove this?

File: pkg/apis/triggers/v1alpha1/trigger_validation_test.go [hunk 4]
```diff
@@ -151,7 +195,7 @@ func Test_TriggerValidate(t *testing.T) {
 						ResourceTemplates: []v1alpha1.TriggerResourceTemplate{{
 							RawExtension: test.RawExtension(t, pipelinev1.PipelineRun{
 								TypeMeta: metav1.TypeMeta{
-									APIVersion: "tekton.dev/v1beta1",
+									APIVersion: "tekton.dev/v1alpha1",
```
Comment (wlynch) [inline pkg/apis/triggers/v1alpha1/trigger_validation_test.go:198]: Is this change intentional?

File: pkg/template/event_test.go [hunk 5]
```diff
@@ -357,73 +358,107 @@ func TestResolveParams(t *testing.T) {
...
-		),
+		template: &triggersv1.TriggerTemplate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "tt-name",
+				Namespace: ns,
+			},
+			Spec: triggersv1.TriggerTemplateSpec{
+				Params: []triggersv1.ParamSpec{{
+					Name:        "param1",
+					Description: "",
```
Comment (wlynch) [inline pkg/template/event_test.go:446]: nit: you can remove fields that just contain the empty value.

**Review by wlynch — COMMENTED** [review]
Mostly LGTM! Few minor things.

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go (hunk 3 above)
Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:618]: yikes, I need to fix this.

File: pkg/apis/triggers/v1alpha1/trigger_validation_test.go (hunk 4 above)
Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/trigger_validation_test.go:198]: nope. Reverting.

File: pkg/template/event_test.go (hunk 5 above)
Comment (dibyom) [reply pkg/template/event_test.go:446]: done.

**Comment by dibyom** [comment]
/hold 

(I missed porting a test case from trigger_binding_Validation.go)

**Review by savitaashture — APPROVED** [review]
